lib/Sema/CSStep.cpp

```diff
@@ -346,8 +346,45 @@ StepResult ComponentStep::take(bool prevFailed) {
 
   /// Try to figure out what this step is going to be,
   /// after the scope has been established.
+  SmallString<64> potentialBindings;
+  llvm::raw_svector_ostream bos(potentialBindings);
+
+  auto bestBindings = CS.determineBestBindings([&](const BindingSet &bindings) {
+    if (CS.isDebugMode() && bindings.hasViableBindings()) {
+      bindings.dump(bos, CS.solverState->getCurrentIndent() + 2);
+    }
+  });
+
   auto *disjunction = CS.selectDisjunction();
-  auto bestBindings = CS.determineBestBindings();
+  auto *conjunction = CS.selectConjunction();
+
+  if (CS.isDebugMode()) {
+    PrintOptions PO;
+    PO.PrintTypesForDebugging = true;
+
+    auto &log = getDebugLogger();
+    if (!potentialBindings.empty()) {
+      log << "(Potential Binding(s): " << '\n';
+      log << potentialBindings;
+    }
+    log.indent(CS.solverState->getCurrentIndent());
+
+    if (disjunction) {
```

Comment:
This and conjunction section prints a selected disjunction/conjunction instead of all of them but let’s address that in a follow up PR instead of holding this one.